sql/src/jmh/java/io/crate/analyze/PreExecutionBenchmark.java

```diff
@@ -27,29 +27,50 @@
 import io.crate.sql.parser.SqlParser;
 import io.crate.sql.tree.Statement;
 import io.crate.testing.SQLExecutor;
-import org.elasticsearch.test.ClusterServiceUtils;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.set.Sets;
 import org.elasticsearch.threadpool.TestThreadPool;
-import org.openjdk.jmh.annotations.Benchmark;
-import org.openjdk.jmh.annotations.Scope;
-import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.*;
 import org.openjdk.jmh.profile.GCProfiler;
 import org.openjdk.jmh.runner.Runner;
 import org.openjdk.jmh.runner.RunnerException;
 import org.openjdk.jmh.runner.options.Options;
 import org.openjdk.jmh.runner.options.OptionsBuilder;
 
 import java.util.UUID;
+import java.util.concurrent.TimeUnit;
 
 @State(value = Scope.Benchmark)
 public class PreExecutionBenchmark {
 
-    // TODO: need to close threadPool
-    private SQLExecutor e = SQLExecutor.builder(
-        ClusterServiceUtils.createClusterService(new TestThreadPool("benchmarks"))
-    ).enableDefaultTables().build();
-    private Statement selectStatement = SqlParser.createStatement("select name from users");
-    private Analysis selectAnalysis = e.analyzer.boundAnalyze(selectStatement, SessionContext.SYSTEM_SESSION, ParameterContext.EMPTY);
-    private UUID jobId = UUID.randomUUID();
+    private TestThreadPool threadPool;
+    private SQLExecutor e;
+    private Statement selectStatement;
+    private Analysis selectAnalysis;
+    private UUID jobId;
+
+    @Setup
+    public void setup() {
+        threadPool = new TestThreadPool("testing");
+        e = SQLExecutor.builder(
+            new ClusterService(Settings.builder().put("cluster.name", "ClusterServiceTests").build(),
```

Comment:
I think that's fine, this is a benchmark not a unittest. I was actually wondering if we should just accept that the threadPool isn't closed if it doesn't influence the benchmarks